File: client/src/app/core/models/view-models/project-view-model.ts
export interface ProjectViewModel {
  id: number;
  key: string;
  name: string;
  description: string;
  repositoryUrl: string;
  workspaceId: number;
  ownerDisplayName: string;
  updatedAt: Date;
  createdAt: Date;
  defaultBoardIdentifier: string;
}

File: server/Netptune.Core/ViewModels/Projects/ProjectViewModel.cs
using System;

namespace Netptune.Core.ViewModels.Projects;

public class ProjectViewModel
{
    public int Id { get; set; }

    public string Key { get; set; } = null!;

    public string Name { get; set; }= null!;

    public string? Description { get; set; }

    public string? RepositoryUrl { get; set; }

    public int WorkspaceId { get; set; }

    public string OwnerDisplayName { get; set; } = null!;

    public DateTime? UpdatedAt { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? DefaultBoardIdentifier { get; set; }

    public string? Color { get; set; }
}
